crates/cdk/src/wallet/receive.rs

```diff
@@ -51,9 +51,29 @@ impl Wallet {
             })
             .collect::<Result<HashMap<String, &String>, _>>()?;
 
-        let p2pk_signing_keys: HashMap<XOnlyPublicKey, &SecretKey> = opts
-            .p2pk_signing_keys
-            .iter()
+        // Build map of X-only pubkey -> SecretKey from stored keys and provided options
+        let mut merged_keys: Vec<SecretKey> = Vec::new();
+        let mut seen: HashSet<XOnlyPublicKey> = HashSet::new();
+
+        // Add keys explicitly provided in options first
+        for sk in &opts.p2pk_signing_keys {
+            let x = sk.x_only_public_key(&SECP256K1).0;
+            if seen.insert(x) {
+                merged_keys.push(sk.clone());
+            }
+        }
+
+        // Merge in any keys stored in the wallet database
+        let stored_keys = self.localstore.list_p2pk_keys().await.unwrap_or_default();
+        for sk in stored_keys {
+            let x = sk.x_only_public_key(&SECP256K1).0;
+            if seen.insert(x) {
+                merged_keys.push(sk);
+            }
+        }
```

Comment:
It seems to me the effect of this is for each receive even ones where no proof needs a signature we get all secret keys from the db and create a list of them, this seem unnecessary. Why don't we iter over the proofs get all pubkeys for those and then get the secret keys for only those pubkeys.